src/movement_controls.py

```diff
@@ -0,0 +1,316 @@
+import arcade
+import math
+from player_input import PlayerInput
+
+# This is setup for the Player to call "drive_input" where the vehicle updates it's
+# intended velocity and rotation based on the player input
+# then the player calls "movement" to act on the change from "drive_input"
+# this way the inputs are always registered but the movement can be skipped
+# if a collision or game state makes it where the vehicle shouldnt act
+# This will help avoid race conditions on collisions
+#
+#
+#
+
+
+class MovementControls:
+
+    # The stat for max (x,y) linear speed
+    drive_speed: float
+    # The stat for max rotational speed
+    turn_speed: float
+
+    # how fast the current acceleration goes from 0 -> 1
+    acceleration_rate: float
+    # how fast the brake applies to current acceleration
+    deceleration_rate: float
+    # how fast the vehicle's current acceleration slows while not accelerating
+    friction: float
+
+    # These are the values the vehicle updates in "drive_input"
+    current_velocity_x: float
+    current_velocity_y: float
+    current_turnVelocity: float
+    current_acceleration: float
+
+    # which control scheme to use
+    vehicle_type: int
+    # for switching between vehicle types
+    toggle: bool
+
+    def __init__(self):
+        self.current_velocity_x = 0
+        self.current_velocity_y = 0
+        self.current_turnVelocity = 0
+        self.current_acceleration = 0
+        self.acceleration_rate = 0.5
+        self.decceleration_rate = 0.3
+        self.brake_rate = 0.3
+        self.friction = 0.1
+        self.drive_speed = 100
+        self.turn_speed = 100
+        self.toggle = False
+        self.vehicle_type = 1
+
+    #   Drive input is called by the player and passed that player's input to set what the vehicle's
+    #    velocity and rotation should be
+    #   Does not change the vehicles position
+    def drive_input(self, delta_time, input, vehicle):
+        # for testing to change between driving types
+        # TODO:find a permanent place for this logic that is not in movement
+        if self.toggle:
+            if input.secondary_fire_button.value == False:
+                self.toggle = False
+        else:
+            if input.secondary_fire_button.value == True:
+                self.toggle = True
+                self.change_vehicle()
+
+        # updates the vehicles intended  velocity and rotation based on type
+        if self.vehicle_type == 0:
+            self.car(delta_time, input, vehicle)
+        elif self.vehicle_type == 1:
+            self.ghost(delta_time, input, vehicle)
+        elif self.vehicle_type == 2:
+            self.ghost_screen_facing(delta_time, input, vehicle)
+        elif self.vehicle_type == 3:
+            self.tank(delta_time, input, vehicle)
+        elif self.vehicle_type == 4:
+            self.mars_walker(delta_time, input, vehicle)
+
```

Comment:
Please implement some sort of indicator on the screen to tell you what driving mode you are in. This could be anything from text that say "car or "ghost" to changing out the car's sprite with a ghost or tank when the driving mode is changed. At least implement for player 1.

Reply:
Adding a HUD ticket for this